kedro-telemetry/kedro_telemetry/plugin.py

```diff
@@ -99,28 +115,90 @@ def before_command_run(
+        default_pipeline = pipelines.get("__default__")
+        hashed_username = _get_hashed_username()
+
+        if not self.project_properties:  # `KedroSession.run` without invoking CLI
+            project_metadata = bootstrap_project(context.project_path)
+            _prepare_project_properties(
+                self.project_properties, hashed_username, project_metadata
+            )
+
+        project_statistics_properties = _format_project_statistics_data(
+            self.project_properties, catalog, default_pipeline, pipelines
+        )
+
+        _send_heap_event(
+            event_name="Kedro Project Statistics",
+            identity=hashed_username,
+            properties=project_statistics_properties,
+        )
+
+
+def _prepare_project_properties(
+    properties: dict, hashed_username: str, project_metadata: ProjectMetadata
 ):
-    """Hash username, format CLI command, system and project data to send to Heap."""
-    hashed_package_name = hashlib.sha512(
-        bytes(project_metadata.package_name, encoding="utf8")
-    ).hexdigest()
-    hashed_project_name = hashlib.sha512(
-        bytes(project_metadata.project_name, encoding="utf8")
-    ).hexdigest()
+
+    hashed_package_name = _hash(project_metadata.package_name)
+    hashed_project_name = _hash(project_metadata.project_name)
     project_version = project_metadata.project_version
 
-    return {
-        "username": hashed_username,
-        "command": f"kedro {' '.join(command_args)}" if command_args else "kedro",
-        "package_name": hashed_package_name,
-        "project_name": hashed_project_name,
-        "project_version": project_version,
-        "telemetry_version": telemetry_version,
-        "python_version": sys.version,
-        "os": sys.platform,
-    }
+    properties.update(
+        {
+            "username": hashed_username,
+            "package_name": hashed_package_name,
+            "project_name": hashed_project_name,
+            "project_version": project_version,
+            "telemetry_version": TELEMETRY_VERSION,
+            "python_version": sys.version,
+            "os": sys.platform,
+        }
+    )
+
+
+def _format_user_cli_data(
+    properties: dict,
+    command_args: List[str],
+):
+    """Add format CLI command data to send to Heap."""
+    cli_properties = properties.copy()
+    cli_properties["command"] = (
+        f"kedro {' '.join(command_args)}" if command_args else "kedro"
+    )
+    return cli_properties
+
+
+def _format_project_statistics_data(
+    properties: dict,
+    catalog: DataCatalog,
+    default_pipeline: Pipeline,
+    project_pipelines: dict,
+):
+    """Add project staitsitcs to send to Heap."""
+    project_statistics_properties = properties.copy()
+    # Initialize default value
+    project_statistics_properties["number_of_datasets"] = 0
+    project_statistics_properties["number_of_nodes"] = 0
+    project_statistics_properties["number_of_pipelines"] = 0
```

Comment:
Why do you need to initialise these values?

Reply:
@MerelTheisenQB I think it's better to have a consistent schema as `default_pipelines` could be empty, in that case, the field will be missing and makes analysis on Heap more difficult.

 Maybe it's better to initialise it to `None` instead, wdyt?
`project_statistics_properties["number_of_nodes"] = len(default_pipeline.nodes) if default_pipeline else None`

Reply:
Yes I think `None` is fine here.